python/ray/serve/logging_utils.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import os
+from typing import Optional
+
+import ray
+from ray.serve.constants import DEBUG_LOG_ENV_VAR, SERVE_LOGGER_NAME
+
+COMPONENT_LOG_FMT = "%(levelname)s %(asctime)s {component_name} {component_id} %(filename)s:%(lineno)d - %(message)s"  # noqa:E501
+LOG_FILE_FMT = "{component_name}_{component_id}.log"
+
+
+def access_log_msg(*, method: str, route: str, status: str, latency_ms: float):
+    """Returns a formatted message for an HTTP or ServeHandle access log."""
+    return f"{method.upper()} {route} {status.upper()} {latency_ms:.1f}ms"
+
+
+def configure_component_logger(
+    *,
+    component_name: str,
+    component_id: str,
+    component_type: Optional[str] = None,
+    log_level: Optional[int] = logging.INFO,
+    log_to_stream: bool = True,
+    log_to_file: bool = True,
+):
+    """Returns a logger to be used by a Serve component.
+
+    The logger will log using a standard format to make components identifiable
+    using the provided name and unique ID for this instance (e.g., replica ID).
+
+    This logger will *not* propagate its log messages to the parent logger(s).
+    """
+    logger = logging.getLogger(SERVE_LOGGER_NAME)
+    logger.propagate = False
+    logger.setLevel(log_level)
+    if os.environ.get(DEBUG_LOG_ENV_VAR, "0") != "0":
+        logger.setLevel(logging.DEBUG)
+
+    formatter = logging.Formatter(
+        COMPONENT_LOG_FMT.format(
+            component_name=component_name, component_id=component_id
+        )
+    )
+    if log_to_stream:
+        stream_handler = logging.StreamHandler()
+        stream_handler.setFormatter(formatter)
+        logger.addHandler(stream_handler)
+
+    if log_to_file:
+        logs_dir = os.path.join(ray.worker._global_node.get_logs_dir_path(), "serve")
+        os.makedirs(logs_dir, exist_ok=True)
+        if component_type is not None:
+            component_name = f"{component_type}_{component_name}"
+        log_file_name = LOG_FILE_FMT.format(
+            component_name=component_name, component_id=component_id
+        )
+        file_handler = logging.FileHandler(os.path.join(logs_dir, log_file_name))
```

Comment:
This is not a rotating file handler, while most of the ray logs are rotated.

Reply:
The ray worker logs are not rotated so not a regression. We can add rotation here if we'd like though